crates/tests/src/rpc/read.rs

```diff
@@ -0,0 +1,1038 @@
+        let decompressed_program = decompress_to_string(contract_program.unwrap());
+
+        let mut class_program_file = File::open("src/rpc/test_utils/class_program.txt").unwrap();
+
+        let mut original_program = String::new();
+        class_program_file.read_to_string(&mut original_program).expect("issue while reading the file");
+
+        let contract_class_file = File::open("src/rpc/test_utils/contract_class.json").unwrap();
+        let reader = BufReader::new(contract_class_file);
+
+        let expected_contract_class: ContractClass = serde_json::from_reader(reader).unwrap();
+
+        let expected_contract_entry_points = match expected_contract_class.clone() {
+            ContractClass::Legacy(compressed) => Some(compressed.entry_points_by_type),
+            _ => None,
+        };
+
+        let expected_contract_abi = match expected_contract_class {
+            ContractClass::Legacy(compressed) => Some(compressed.abi),
+            _ => None,
+        };
+
+        assert_eq!(decompressed_program, original_program);
+        assert_eq!(contract_entry_points, expected_contract_entry_points);
+        assert_eq!(contract_abi, expected_contract_abi);
+    }
+
+    /*
+       curl --location 'https://free-rpc.nethermind.io/sepolia-juno/' \
+       --header 'Content-Type: application/json' \
+       --data '{
+           "jsonrpc": "2.0",
+           "method": "starknet_estimateFee",
+           "params": {
+               "request": [
+                   {
+                       "type": "DEPLOY_ACCOUNT",
+                       "version": "0x1",
+                       "max_fee": "0x0",
+                       "signature": [
+                           "0x73d0a8a69f0ebf44b1c2bb2a9e85bf998883eb2008ca7b9c57b6f28dacb6dd8",
+                           "0x4a43711cd08f55ef73603f1e7b880c7f438fb68934f0823a736f9f577ab040a",
+                           "0x0",
+                           "0x0",
+                           "0x0",
+                           "0x0",
+                           "0x0",
+                           "0x0",
+                           "0x0",
+                           "0x0"
+                       ],
+                       "nonce": "0x0",
+                       "contract_address_salt": "0x0",
+                       "constructor_calldata": ["0x2e23f1647b018bfb3fe107e2ebd4412f0a0ed41bd60c10d842a76f8cdbbe1ba"],
+                       "class_hash": "0x05c478ee27f2112411f86f207605b2e2c58cdb647bac0df27f660ef2252359c6"
+                   }
+               ],
+               "simulation_flags": [ "SKIP_VALIDATE"],
+               "block_id": {
+                   "block_number": 1
+               }
+           },
+           "id": 1
+       }'
+
+       {"jsonrpc":"2.0","result":[{"gas_consumed":"0xcfc","gas_price":"0x3b9ada0f","data_gas_consumed":"0x0","data_gas_price":"0x1","overall_fee":"0x305eea75ac4","unit":"WEI"}],"id":1}%
+    */
+    #[ignore]
+    #[tokio::test]
+    async fn test_estimate_fee_works() {
```

Comment:
wait so are these working?

Reply:
added ignore on these, the data might be helpful in case we test it again